Sources/AblyChat/RoomStatus.swift

```diff
@@ -1,17 +1,57 @@
 import Ably
 
-// TODO: rename
+/**
+ * The different states that a room can be in throughout its lifecycle.
+ */
 public enum RoomStatus: Sendable, Equatable {
+    /**
+     * A temporary state for when the room object is first initialized.
+     */
     case initialized
+
+    /**
+     * The library is currently attempting to attach the room.
+     */
     case attaching(error: ARTErrorInfo?)
+
+    /**
+     * The room is currently attached and receiving events.
+     */
     case attached
+
+    /**
+     * The room is currently detaching and will not receive events.
+     */
     case detaching
+
+    /**
+     * The room is currently detached and will not receive events.
+     */
     case detached
+
+    /**
+     * The room is in an extended state of detachment, but will attempt to re-attach when able.
+     */
     case suspended(error: ARTErrorInfo)
+
+    /**
+     * The room is currently detached and will not attempt to re-attach. User intervention is required.
+     */
     case failed(error: ARTErrorInfo)
+
+    /**
+     * The room is in the process of releasing. Attempting to use a room in this state may result in undefined behavior.
+     */
     case releasing
+
+    /**
+     * The room has been released and is no longer usable.
+     */
     case released
 
+    // TODO: remove this tmp comment:
```

Comment:
@lawrence-forooghian could you please take a look

Reply:
cc @ttypic

Reply:
That property exists in the JS SDK because it's the only way to find the error that's associated with a room status. In d51887c67fd3c718e1474bc05aeeb947b5c431c9, when I moved the error into an associated value of the `RoomStatus` enum, I removed this property. If there's a compelling reason to restore it, we can, but without having a good example it's not clear to me why a user of the library would want to fetch the error associated with a `RoomStatus` without caring about the enum case too.

Reply:
I think I've noticed that while were writing README: https://github.com/ably/ably-chat-swift/blob/f95526009f12e84a9cf30e8fb576b5c2c6a9592d/README.md?plain=1#L203

So I decided to leave it the same as in the kotlin readme (as I think it has a little sense to keep difference between two libraries in this particular place) and to raise this question later, which I did.

Reply:
IMHO it is, as I think that all APIs should be as close to each other as possible.

Reply:
I disagree. Similarity does not mean or imply absolute equality. The readme should be updated to accurately document the Swift solution without caring about Kotlin 

(especially given all work around the public APIs and any differences between them should have been resolved in other PRs)

Reply:
[updated](https://github.com/ably/ably-chat-swift/pull/165/commits/02977daf814e783a6b8fc38616d8ddc3d30a0241)

Reply:
In addition to that in kotlin this property is within a room itself (not status). Which doesn't make it any better for this situation, bc our `Room` doesn't have it either. wdyt @umair-ably @lawrence-forooghian

Reply:
Updated readme https://github.com/ably/ably-chat-swift/pull/166#issuecomment-2532149830

Reply:
I'm not sure I follow... we do have a `RoomStatus` object on a `Room` object in swift

Reply:
Sure, but here we are talking about the `error`.

Reply:
yes, via a status

Reply:
The question was whether we should adopt the sdk to this or not. Decided that we shouldn't, so this can be resolved. I've updated both repos.

Reply:
Also I've removed `status.error` without showing how to get the error from the status to not overwhelm user reading README with all possible switch cases with an error. But now I think that something like this can be added:
```
switch try await room.status {
   case let .failed(error: error):
      let _ = error
   }
   ...
}
```
Let me know if this is suitable.

Reply:
Also we have `connection.error`, but also this - https://github.com/ably/ably-chat-swift/issues/12
Looks like optional error will be removed also.

Reply:
nice one, yeah i think that example is suitable. 

Slightly annoying that connection error does differ atm, but we'll likely go with the same solution there by the time we get to GA

Reply:
@umair-ably @lawrence-forooghian done - https://github.com/ably/ably-chat-swift/compare/917c94ca39b498cffc5155d420bdcf05c48352ec..49360b63fe22c18f2a37a5e651adcb9ea86cf905